realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -307,6 +307,27 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(Required.class) == null) {
+                    // The field doesn't have the @Required annotation
```

Comment:
The comment and what is actually happening seems a bit strange.The comment just says what the if statement does (= not needed), but then we only save some fields in the nullableElements set. Better to describe why that is.